docs/src/architecture/data-model.md

```diff
@@ -0,0 +1,348 @@
+
+# thin-edge data-model
+
+The **thin-edge data-model** is used to represent all device-related information.
+It covers the thin-edge device itself, as well as other _external devices_ connected to thin-edge.
+Each device is represented with a data **object**. 
+Each data **object** can have different **fields**. 
+The following JSON code shows a small example of a device data **object**:
+       
+```json
+          {
+              "name": "thin-edge device",
+              "type": "thin-edge.io"
+          }
+```
+
+**thin-edge** manages an **inventory** to store and provide all those objects. 
+
+The figure below illustrats the **data-model** objects and the **inventory**:
+
+![thin-edge Inventory](images/inventory.svg)
+
+* The **thin-edge Device** object represents the device it-self, that runs **thin-edge** and manages that **inventory**.
+  That objects has the fields `name` and `type` that contains the device-name and device-type visible in the cloud.
+
+* The **Telemetry Descriptor** object is part of the **thin-edge Device** object 
+  and contains a description of all **metrics**, **commands**, **events** and **alarms** the device provides.
+  * For all of those telemetry data the **Telemetry Descriptor** object contains individual descriptions:
+    ```json
+       "telemetry_descriptor": {
+          "metrics":  { /* ... specific description to metrics ...   */ },
+          "commands": { /* ... specific description to commands ...  */ },
+          "events":   { /* ... specific description to events ...    */ },
+          "alarms":   { /* ... specific description to alarms ...    */ },
+       } 
+    ```
+  * **Metrics**: Contains all information about metrics and measurements<br/>
+    Example:
+    ```json
+       "metrics": {
+            "weather_station": {
+                "num_values": 2,
+                "units": ["%", "celsius"] // humidity and temperature
+            },
+            "power_meter": {
+                "num_values": 3,
+                "units": ["V", "V", "V"] // voltage of phase 1, 2, 3
+            },
+
+            /* general structure for a metric: */
+            "<type_name>": {
+                "num_values": <number of values>,
+                "units": ["<unit of 1st value>", "<unit of 2nd value>", ...]
+            }
+
+        }
+     ```
+      * Where
+        * `type_name`, is a reference string, unique in scope of the given device object
+        * `num-values`, number of values the metric's measurement carries
```

Comment:
The `num-values` field is not required once the TODO item is implemented. Or is this more of a conviniece property instead of just counting the length of the series array?